packages/flutter_tools/lib/src/http_host_validator.dart

```diff
@@ -59,71 +57,60 @@ class HttpHostValidator extends DoctorValidator {
     _platform.environment[kEnvCloudUrl] ?? kgCloudHttpHost,
   ];
 
-  /// Make a head request to the HTTP host. HTTP Host is available if no exception happened
+  /// Make a head request to the HTTP host for checking availability
   Future<_HostValidationResult> _checkHostAvailability(String host) async {
+    late final int timeout;
     try {
-      final int timeout =
-      int.parse(_platform.environment[kDoctorHostTimeout] ?? '10');
+      timeout = int.parse(_platform.environment[kDoctorHostTimeout] ?? '10');
```

Comment:
can this also be lifted outside of the try block?

Reply:
See #98328.

Reply:
Are you catching `FormatException`?

Reply:
Not in this PR, but in the PR that fixes that issue, and I see no point in moving this line when eventually it has to be moved back.

Reply:
any reason not to catch and toolexit on formatexception in this PR?

Reply:
Nothing specific, thought it would be better to land separate PRs to fix separate issues, but I can include all in one if you are okay with that.

Reply:
Sounds good to me as long as it's not a lot more code, which I imagine it wouldn't be?

Reply:
@christopherfujino I've pushed a commit that handles all possible exceptions, PTAL.